### PR TITLE
Non-functional change: refactor of collections routing/component code

### PR DESCRIPTION
**Pull request by alexbfree**
This is preparation for the new project-context menu #2253 

This PR just refactors the collections & favorites routes and components such that navigation and data listing are separated out, and collections and favorites are separated out, and URLs are defined with a base href.

Deployed to staging at https://maintain-project-context-v2.pfe-preview.zooniverse.org/

Please leave branch open after merge.

### CODE REVIEW

**Comment by alexbfree** [comment]
Hi @brian-c , please could you review/merge this. As we talked about last week, I have split up owned-card list and moved collections to separate components as a first step towards the project context menu approach we discussed. (Am just coming back to this after putting it down to work on other things last week).

**Comment by alexbfree** [comment]
@brian-c I have gone one further and completely removed Owned Card List (Which I hope you will be pleased about as I know you hated it. 

I have updated https://maintain-project-context-v2.pfe-preview.zooniverse.org/ right now with new code from this branch.

Still a non-functional change - only pages that could have been affected are Home (at bottom), Projects, and Collections page (both single user route and all users). All tested and still working.

File: app/collections/show.cjsx [hunk 1]
```diff
@@ -41,23 +41,23 @@ CollectionPage = React.createClass
     <PromiseRenderer promise={@props.collection.get('owner')}>{(owner) =>
       [ownerName, name] = @props.collection.slug.split('/')
       params = {owner: ownerName, name: name}
-
+      baseLink = "/collections/#{ownerName}/#{name}"
```
Comment (brian-c) [inline app/collections/show.cjsx]: `params` above is unused, which means `ownerName` and `name` are unused except for here, where it'd be more appropriate to use `@props.collection.slug` anyway.

File: app/partials/project-card.cjsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  getDefaultProps: ->
+    avatar: ''
+    name: ''
+    slug: ''
+
+  componentDidMount: ->
+    card = @refs.projectCard
+
+    Promise.resolve (if @props.project.avatar_src then "//#{ @props.project.avatar_src }" else '/assets/simple-avatar.jpg')
```
Comment (brian-c) [inline app/partials/project-card.cjsx]: What's going on here? Why are we making a promise for a value we already have?

File: app/partials/project-card.cjsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      .then (src) =>
+        card.style.backgroundImage = "url('#{src}')"
+        card.style.backgroundSize = "contain"
+      .catch =>
+        card.style.background = "url('/assets/simple-pattern.jpg') center center repeat"
+
+    card.classList.add 'project-card'
```
Comment (brian-c) [inline app/partials/project-card.cjsx]: Why don't we apply these styles and this class in the `render` method?

**Comment by brian-c** [comment]
A couple nitpicks, probably leftover from old code, otherwise solid.

File: app/partials/project-card.cjsx (hunk 2 above)
Comment (alexbfree) [inline app/partials/project-card.cjsx]: I am not sure. This is pre-existing code I just moved across from `partials/owned-card.cjsx`. I can investigate, though this question shouldn't necessarily delay this PR.

File: app/partials/project-card.cjsx (hunk 3 above)
Comment (alexbfree) [inline app/partials/project-card.cjsx]: again, from old code. yes i can look at that.

File: app/collections/show.cjsx (hunk 1 above)
Comment (alexbfree) [inline app/collections/show.cjsx]: oops yes I should have spotted that. will delete.

**Comment by alexbfree** [comment]
Thanks for taking a look. OK, yes you're right, mostly issues from old code. I will clean them up and update this PR then notify you.

**Comment by alexbfree** [comment]
@brian-c I have added a new commit to address your review comments. Should be ready for you to merge if you are happy.

**Comment by alexbfree** [comment]
just rebased to latest zoo master

**Pull request merged**